LinuxAdapter.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.DirectoryServices.Protocols;
+using System.Linq;
+using System.Net;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Authentication.Negotiate
+{
+    public static class LinuxAdapter
+    {
+        // example: machineAccount = "user@DOMAIN.com" machinePassword = "***"
+        public static Task OnAuthenticated(AuthenticatedContext context, string machineAccount, string machinePassword, bool resolveNestedGroups = true)
```

Comment:
Is one of the reasons this is so slow that it sets up a new connection per-reqeuest? Make LinuxAdapter non-static, pass these parameters to the constructor, and have it re-use the connection. That's more representative of the intended design.

Reply:
Setting up a new connection per request isn't what I found to be slow, that's mostly instantaneous. Turns out the slowness is likely just my environment since it wasn't reproduced in Jose's environment. I haven't tracked down what my slowness is caused by though.

Reply:
Regardless, we shouldn't be doing a new connection per-request.

Reply:
Depends in connections are thread safe though? We've had issues before in sharing things where results got mixed up

Reply:
I was assured they are, or else we're going to have a whole different challenge trying to use these APIs.

Reply:
Yup @joperezr said they were